### PR TITLE
Upgraded packages and dependecy fixes

### PR DESCRIPTION
**Pull request by vivek-kumar-pandey**
#456 

### Types of changes

- [ ] Bug fix (a change which fixes an issue)
- [x] New feature (a change which adds functionality)
- [ ] Unit test (a unit test that partially or fully covers sanity and code coverage of a functionality)

### Short description of what this resolves:

We needed to fix upgrade the Flask version which required some dependency upgrades too. Also we needed to fix some issues which were coming after the upgrade:
- [x] Upgraded dependent packages
- [x] Fixed BaseQuery class issue
- [x] Fixed Unit test case error
- [x] Fixed the SQLAlchemy Track modification error
- [x] Fixed error related to instance not found

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [x] The code changes are passing the CI checks
- [ ] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: api/services/prompt_service.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # This file is treated as service layer
-from api import models, db
+from api import db
 from datetime import datetime
 from api.helpers import prompt_helper, common_helper, db_helper
 from utils.loggingutils import logger
+from api import models
```
Comment (Sachinbisht27) [inline api/services/prompt_service.py:6]: We can revert this change if `db` and `models` both are imported from `api`
```suggestion
from api import models, db
from datetime import datetime
from api.helpers import prompt_helper, common_helper, db_helper
from utils.loggingutils import logger
```

Comment (vivek-kumar-pandey) [reply api/services/prompt_service.py:6]: @Sachinbisht27 I was getting circular import errors due to defining them together, but importing like this solved the issue.

File: api/models/user_custom_fields.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from api.mixins import TimestampMixin
 from api import db
-from flask_sqlalchemy import BaseQuery
+
```
Comment (Satendra-SR) [inline api/models/user_custom_fields.py:3]: Please remove extra space

File: config.py [hunk 3]
```diff
@@ -48,7 +48,6 @@
 if github_action_db_url:
     SQLALCHEMY_DATABASE_URI = github_action_db_url
 
-SQLALCHEMY_TRACK_MODIFICATIONS = True
```
Comment (Satendra-SR) [inline config.py:51]: What was the reason behind removing this config?

**Review by Satendra-SR — COMMENTED** [review]
@vivek-kumar-pandey Added a few questions. 
Also, can we add some specific details in the PR description please?

File: requirements.txt [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
 chardet==4.0.0
-click==7.1.2
-Flask==1.1.2
+click==8.1.3
+Flask==3.0.0
 requests==2.25.1
 python-dotenv==0.15.0
-Flask-SQLAlchemy==2.1
+Flask-SQLAlchemy==3.1.1
+greenlet==3.0.3
```
[the remark this answers is not in the document]
Comment (vivek-kumar-pandey) [reply requirements.txt:8]: While upgrading the flask sqlalchemy, greenlet was also installed as dependency. So i added it to the requirement just to make sure that it is not missed in the staging or production.

File: config.py (hunk 3 above)
Comment (vivek-kumar-pandey) [reply config.py:51]: I was getting error related to SQLALCHEMY_TRACK_MODIFICATIONS which got fixed after removing this.

File: api/helpers/db_helper.py [hunk 5]
```diff
@@ -20,7 +20,7 @@ def get_class_by_tablename(tablename):
     :param tablename: String with name of table.
     :return: Class reference or None.
     """
-    for classObject in db.Model._decl_class_registry.values():
+    for classObject in db.Model.registry._class_registry.values():
```
Comment (vivek-kumar-pandey) [inline api/helpers/db_helper.py:23]: The _decl_class_registry was throwing error as in the latest version it is removed from the Model object. So updated it accordingly

File: api/models/call_log.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 from api.mixins import TimestampMixin
 from api import db
-from flask_sqlalchemy import BaseQuery
+from flask_sqlalchemy.query import Query as BaseQuery
```
Comment (vivek-kumar-pandey) [inline api/models/call_log.py:3]: The Basequery class is now named as Query and is placed within the query module

File: tests/conftest.py [hunk 7]
```diff
@@ -27,21 +27,21 @@ def client(app):
 
 @pytest.fixture(scope="session")
 def db():
-    db = SQLAlchemy(flask_app)
-    db.init_app(flask_app)
+    db = SQLAlchemy()
```
Comment (vivek-kumar-pandey) [inline tests/conftest.py:30]: Previously It was throwing an error while running unit test, that the SQLAlchemy instance was already defined

**Review by Satendra-SR — APPROVED** [review]
